### PR TITLE
Firebase Authentication + Firestore setup

### PR DESCRIPTION
**Pull request by nadinekuo**
- Firebase Authentication setup (dummy users added for now)

image

- Basic login screen that allows users to login - see `.../frontend/README` for dummy credentials you can use for logging in (we may make it more fancy later once we have a logo etc.)

<p align="center">

</p>

- Basic logout functionality under "Settings" screen for verifying Firebase Auth (issue #38 handles this screen)

- Firestore database setup
image

- Data models for User and Field added (more models and fields can be added on basis of necessity in the future)

- Very simple list of fields displayed under "My Fields" screen to verify database interaction (issue #16 will handle this)

### CODE REVIEW

File: src/frontend/lib/views/myfields/my_fields.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            onPressed: () async {
+              await Navigator.push(
+                  context,
+                  MaterialPageRoute(
+                    builder: (context) => const AddField(),
+                  )
+              );
+            },
+            tooltip: 'Add field',
+            child: const Icon(Icons.add, size: 35, color: Colors.white,),
```
Comment (DimanSUzunov) [inline src/frontend/lib/views/myfields/my_fields.dart:50]: We would have to think how to make icons and elements responsive in the future

File: src/frontend/lib/views/myfields/fields_list.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
```
Comment (DimanSUzunov) [inline src/frontend/lib/views/myfields/fields_list.dart:1]: Is Fields List a component for all fields that exist, and MyFields for the fields of the user?

**Comment by DimanSUzunov** [comment]
The setup looks good, and the structure as well. Could you link the resource you used to do the authentication?

**Review by DimanSUzunov — APPROVED** [review]

File: src/frontend/lib/views/myfields/fields_list.dart (hunk 2 above)
Comment (nadinekuo) [reply src/frontend/lib/views/myfields/fields_list.dart:1]: No, FieldList is an inner widget of the MyFields widget, listing all fields of the user

**Pull request merged**